keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1170 @@
+- Modify the "Allocate" static policy to check for the option, align-cpus-by-uncorecache. For platforms where SMT is enabled, align-cpus-by-uncorecache will typically take full cores and also works with the option full-pcpus-only.
+- Align-cpus-by-uncorecache is not compatible with the options DistributeCPUsAcrossNUMAOption and DistributeCPUsAcrossCoresOption.
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- Feature is enabled by a static policy option flag.  
+- It does not change behavior of non-static policy
+- It preserves the behavior of other static options.
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+- Add `UnCoreCacheID` in `CPUInfo` and modify CPUTopology struct to also track uncorecaches in `topology.go`
+  
+  ```go
+  // CPUTopology contains details of node cpu, where :
+  // CPU  - logical CPU, cadvisor - thread
+  // Core - physical CPU, cadvisor - Core
+  // Socket - socket, cadvisor - Socket
+  // NUMA Node - NUMA cell, cadvisor - Node
+  // UnCoreCache - Split L3 Cache Topology, cadvisor
+  type CPUTopology struct {
+  NumCPUs        int
+  NumCores       int
+  NumUnCoreCache int
+  NumSockets     int
+  NumNUMANodes   int
+  CPUDetails     CPUDetails
+  }
+  type CPUInfo struct {
+    NUMANodeID int
+    SocketID   int
+    CoreID     int
+    UnCoreCacheID  int
+  }
+  ```
+  
+- Get the UnCoreCacheID for each CPU from cadvisor.
+
+  ```go
+  func Discover(machineInfo *cadvisorapi.MachineInfo) (*CPUTopology, error) {
+  CPUDetails[cpu] = CPUInfo{
+    CoreID:     coreID,
+    SocketID:   core.SocketID,
+    NUMANodeID: node.Id,
+    UnCoreCacheID:  core.UncoreCaches[0].Id
+    }
+  }
+    ```
+
+- Enabling the cpu manager policy option in kubelet.yaml
```

Comment:
I would suggest restarting this and just define the Kubelet Config field that you want to add.

Reading this, it sounds like this field already exists but you are proposing a new KubeletConfig option called "align-cpus-by-uncorecache".

Reply:
I think this refers to the `CPUManagerPolicyOptions` https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2625-cpumanager-policies-thread-placement#graduation-criteria-of-options . We can however use a bit of clarification/rephrasing

Reply:
@ajcaldelas Let's also note in the readme to set topology-manager-policy to "single-numa-node" on multiple sockets system to prevent cross-die assignments